src/variant.cpp

```diff
@@ -526,7 +542,11 @@ __uint128 variant::to_uint128() const {
     if( is_uint128() || is_int128()) {
         return value_.as_uint128;
     } else if (is_string()) {
+#ifndef ABSENT__INT128_TYPEID
        return boost::lexical_cast<__int128>(value_.as_string);
+#else
+       return boost::lexical_cast_128(value_.as_string);
+#endif
```

Comment:
may be better to hide this inside boost::lexical_cast_128(value_.as_string);
And only call it there.

Reply:
agree